### PR TITLE
Cop 10943 - add newPageId prop

### PR DESCRIPTION
**Pull request by fisherrj2**
Relates to ticket: https://support.cop.homeoffice.gov.uk/browse/COP-11202

Additions made to form renderer to allow viewing a submitted tasklist form that is broken down by its individual tasks following a similar pattern to check your answers and allowing the action links to be hidden.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@ukhomeoffice/cop-react-form-renderer",
-  "version": "3.10.0",
+  "version": "3.10.1",
```
Comment (aliktb) [inline package.json:3]: I think this should be a minor change, not a patch so `3.11.0`

File: src/json/group.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
       "label": "First name",
       "type": "input",
       "readonly": true,
-      "groupId": "addressDetails",
+      "groupid": "addressDetails",
```
Comment (aliktb) [inline src/json/group.json:14]: Just wondering why these have changed to lowercase id?

File: src/components/FormRenderer/FormRenderer.jsx [hunk 3]
```diff
@@ -103,6 +106,10 @@ const InternalFormRenderer = ({
     setPages(Utils.FormPage.getAll(_pages, components, { ...data }));
   }, [components, _pages, data, setPages]);
 
+  useEffect(() => {
+    if (newPageId !== undefined) onPageChange(newPageId);
+  },[newPageId]);
```
Comment (aliktb) [inline src/components/FormRenderer/FormRenderer.jsx:111]: Github actions is saying that `onPageChange` should be part of the dependency array so:

```
useEffect(() => {
    if (newPageId !== undefined) onPageChange(newPageId);
  },[newPageId, onPageChange]);
```

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx [hunk 4]
```diff
@@ -99,15 +101,53 @@ const CheckYourAnswers = ({
...
     <div className={DEFAULT_CLASS}>
       {title && !hide_title && (
         <LargeHeading key='heading'>{title}</LargeHeading>
       )}
       {errors && errors.length > 0 && <ErrorSummary errors={errors} />}
-      {pages &&
+      {type === 'task-list-cya' && pages.length > 0 && sections &&
+        sections.map((section) => {
+          return section.tasks.map((task) => {
+            let filterPages = pages.filter(function (page) {
```
Comment (aliktb) [inline src/components/CheckYourAnswers/CheckYourAnswers.jsx:115]: could we make this an arrow function. Just to keep things consistent with the rest of form renderer

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx [hunk 5]
```diff
@@ -99,15 +101,53 @@ const CheckYourAnswers = ({
...
+                {(
+                  <>
+                    <h3 className="govuk-heading-m govuk-!-margin-top-8 govuk-!-margin-bottom-0">{filterPages.length > 0 && task.name}</h3>
+                    {filterPages.map((page, pageIndex, array) => {
+                      let hideActionButtons;
+                      isGroup(page.id) ? hideActionButtons = true : hideActionButtons = noChangeAction;
+                      return (
+                        <Fragment key={pageIndex}>
+                          {(!hide_page_titles && page.title && !isGroup(page.id)) && (
+                            <>
```
Comment (aliktb) [inline src/components/CheckYourAnswers/CheckYourAnswers.jsx:131]: do we need a fragment here?

File: src/components/CheckYourAnswers/CheckYourAnswers.test.js [hunk 6]
```diff
@@ -168,6 +168,48 @@ describe('components', () => {
...
+          name: 'These are your tasks',
+          tasks: [
+            { name: 'Task 1', state: 'complete', pages: ['names'] },
+            { name: 'Task 2', state: 'complete', pages: ['grade'] },
+          ],
+        },
+      ];
+      await act(async () => {
+        renderDomWithValidation(
+          <CheckYourAnswers pages={PAGES} onRowAction={ON_ROW_ACTION} onAction={ON_ACTION} hide_title={true} sections={sections} type={"task-list-cya"} />,
```
Comment (aliktb) [inline src/components/CheckYourAnswers/CheckYourAnswers.test.js:183]: I don't think we need {} around `"task-list-cya"`

**Review by aliktb — COMMENTED** [review]
left a few comments

File: package.json (hunk 1 above)
Comment (fisherrj2) [reply package.json:3]: Updated to 3.11.0

File: src/json/group.json (hunk 2 above)
Comment (fisherrj2) [reply src/json/group.json:14]: It was to resolve a separate preexisting warning about using groupId - recommendation was to use all lower case

File: src/components/FormRenderer/FormRenderer.jsx (hunk 3 above)
Comment (fisherrj2) [reply src/components/FormRenderer/FormRenderer.jsx:111]: Added to dependency array

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx (hunk 4 above)
Comment (fisherrj2) [reply src/components/CheckYourAnswers/CheckYourAnswers.jsx:115]: Arrow function now used

File: src/components/CheckYourAnswers/CheckYourAnswers.test.js (hunk 6 above)
Comment (fisherrj2) [reply src/components/CheckYourAnswers/CheckYourAnswers.test.js:183]: Removed

File: src/components/CheckYourAnswers/CheckYourAnswers.jsx (hunk 5 above)
Comment (fisherrj2) [reply src/components/CheckYourAnswers/CheckYourAnswers.jsx:131]: No and it has been removed

File: package.json (hunk 1 above)
Comment (fisherrj2) [reply package.json]: May now need to be 3.12.0?

File: src/components/CheckYourAnswers/CheckYourAnswers.test.js [hunk 7]
```diff
@@ -168,6 +168,48 @@ describe('components', () => {
...
+          name: 'These are your tasks',
+          tasks: [
+            { name: 'Task 1', state: 'complete', pages: ['names'] },
+            { name: 'Task 2', state: 'complete', pages: ['grade'] },
+          ],
+        },
+      ];
+      await act(async () => {
+        renderDomWithValidation(
+          <CheckYourAnswers pages={PAGES} onRowAction={ON_ROW_ACTION} onAction={ON_ACTION} hide_title={true} sections={sections} type="task-list-cya" />,
```
Comment (aliktb) [inline src/components/CheckYourAnswers/CheckYourAnswers.test.js:183]: one little thing I've just noticed. I think you can just do `hide_title`. I don't think there is a need for `{true}`

Comment (fisherrj2) [reply src/components/CheckYourAnswers/CheckYourAnswers.test.js]: done

**Review by aliktb — APPROVED** [review]

**Pull request merged**